tests/check_framework/tests.py

```diff
@@ -303,5 +303,25 @@ class ModelWithDescriptorCalledCheck(models.Model):
 
 
 class ChecksRunDuringTests(SimpleTestCase):
+    databases = '__all__'
     def test_registered_check_did_run(self):
-        self.assertTrue(my_check.did_run)
+        default = connections['default']
+        settings_dict = default.settings_dict
+        old_name = settings_dict['NAME']
+        check_last = old_name[-2:]
+        try:
+            if default.vendor != 'sqlite' and int(check_last[1]):
+                settings_dict['NAME'] = old_name[:-2]
+        except ValueError:
+            self.assertTrue(my_check.did_run)
+        # Redirect settings to check model creation in original database
+        default.settings_dict.update(settings_dict)
+        default.close()
+        default.connect()
+        check_ran = CheckModel.objects.exists()
+        settings_dict['NAME'] = old_name
+        default.close()
+        default.settings_dict.update(settings_dict)
+        default.connect()
+        self.assertTrue(check_ran)
```

Comment:
Why doesn't the row get copied in the database dump, along with others like the migrations and content types? Don't the system checks run before the new test processes and the database cloning?

Reply:
The system checks run after we setup the databases, including their clones. So this row does not exist in cloned databases.

We could rewrite it so cloning happens after system checks are run and sidestep this ugly bit of code, and I'm personally leaning towards that. I'm currently cloning SQLite after system checks are run because I couldn't find a good way to redirect the connection for this test.

Reply:
> We could rewrite it so cloning happens after system checks are run

If that's easy enough I think it's a good idea. It means that check errors will be displayed and stop the test process sooner, speeding up feedback.

Reply:
Could we possibly run checks before setup'ing test databases? What are the benefits of running checks on test databases in the first place?

Reply:
Some checks require database access e.g. mysql.W002 ( https://docs.djangoproject.com/en/3.0/ref/checks/#database ) , django-mysql's checks ( https://django-mysql.readthedocs.io/en/latest/checks.html ).

Those I've linked to don't strictly need access to an existing schema but I think changing this could be considered a breaking change.

Reply:
@adamchainz I've taken a stab at this and added a ```call_command('check', ....)``` call inside ```setup_databases```.

Motivation is splitting up that adding a ```clone_databases``` doesn't gel well with the [timings patch](https://github.com/django/django/pull/13224) and there isn't much justification for adding it other than the check_framework test failure.

With that in mind, I suggest two possible paths to move forward with this discussion:
- Add a ```call_command('check', ...)``` inside setup_databases (possibly breaking tests and expected behavior)
- Rewrite the check_framework test

I'm currently exploring option 1, but I think it might be more trouble than it's worth and introduce more headaches because the test runner's code is written using the assumption that setup_databases would never handle exceptions raised by running checks.

What do you think?